javascript/code-challenges/challenges-13.test.js

```diff
@@ -65,7 +75,7 @@ Write a function named allHappy that takes in an array of strings and returns a
 ------------------------------------------------------------------------------------------------ */
 
 const allHappy = (arr) => {
-  // Solution code here...
+  return arr.reduce((str) => str.includes(":)"));
 };
 
```

Comment:
Hello again!

Ok, so for this one, good approach using reduce() as we want to return a single value, a boolean.

 With reduce it is always helpful to think about the two callbacks that are used, current and accumulator. The `curr` is always the current item we are iterating over at that time, the `acc `is the total value or accumulated value that we carry with us throughout each iteration ( I know, whew, that is a lot to digest).

So inside your function `allHappy()` let's break some things down a bit:
- When using reduce, first think about the initial value. What is it here? Well, we want to set it to be a boolean and the base set up of function would be:
```
const allHappy = arr => {
const stringSearch = arr.reduce((acc, curr) => {
// code here
},true);
// possible return of stringSearch
};
```
- How do we handle the current element or item? We can use `includes()` to say if current item does not include the ':)' then return false (or vice-versa)
- else what do we need to do? Well, with reduce, we always have to eventually return the `acc`. So perhaps we can return the acc if it equals true only! Else return false.
- Once all is said and done here, after we return acc, we may need to return the variable holding the value that reduce gives us 
Keep in mind, this is one approach :) I hopes this helps a bit 👍